third_party/proton/csrc/lib/Profiler/CuptiProfiler.cpp

```diff
@@ -12,6 +12,42 @@ namespace proton {
 
 namespace {
 
+struct CuptiState {
+  CuptiProfiler *profiler;
+  std::set<Data *> dataSet;
+  size_t level{0};
+  bool isRecording{false};
+  Scope scope{};
+
+  void record(const Scope &scope, CuptiProfiler *profiler) {
+    this->scope = scope;
+    this->profiler = profiler;
+    this->dataSet = profiler->getDataSet();
+  }
+
+  void reset() {
+    dataSet.clear();
+    level = 0;
+    scope = Scope();
+  }
+
+  void enterOp() {
+    profiler->enterOp(scope);
+    for (auto data : dataSet) {
+      data->enterOp(scope);
+    }
+  }
+
+  void exitOp() {
+    profiler->exitOp(scope);
+    for (auto data : dataSet) {
+      data->exitOp(this->scope);
+    }
+  }
+};
+
+static inline thread_local CuptiState cuptiState;
```

Comment:
what does `inline` mean here, since this is a cpp file?

Reply:
haha, forgot to delete...